File: notifier/src/main/java/com/juanimoli/strictmodenotifier/detector/NetworkDetector.java
package com.juanimoli.strictmodenotifier.detector;

import com.juanimoli.strictmodenotifier.StrictModeLog;

public class NetworkDetector implements Detector {
    @Override
    public boolean detect(StrictModeLog log) {
        return log.message.contains("StrictMode$StrictModeNetworkViolation");
    }
}

File: notifier/src/main/java/io/github/juanimoli/strictmode/notifier/detector/NetworkDetector.kt
package io.github.juanimoli.strictmode.notifier.detector

import io.github.juanimoli.strictmode.notifier.StrictModeLog

class NetworkDetector : Detector {
    private val detectMessageList = listOf(
        "StrictMode(.*)StrictModeNetworkViolation".toRegex(),
        "StrictMode(.*)NetworkViolation".toRegex()
    )

    override fun detect(log: StrictModeLog): Boolean {
        return detectMessageList.any { it.find(log.message) != null }
    }
}
